Design/AlignDetectors_MoreRework.md

```diff
@@ -0,0 +1,51 @@
+* focus ([DiffractionFocussing](https://docs.mantidproject.org/nightly/algorithms/DiffractionFocussing.html)). Focus the data for each bank of detectors according to grouping data in a .cal file
+* convert data back to TOF ([ConvertUnits](https://docs.mantidproject.org/nightly/algorithms/ConvertUnits-v1.html))
+* Save data to file for downstream processing ([SaveGSS](https://docs.mantidproject.org/nightly/algorithms/SaveGSS-v1.html))
+
+I believe there is a similar workflow followed at SNS using the [AlignAndFocusPowder](https://docs.mantidproject.org/nightly/algorithms/AlignAndFocusPowder-v1.html) workflow algorithm. ISIS use some python scripts to chain together the different steps but it achieves a similar process to AlignAndFocusPowder
+
+The problem is that the final ConvertUnits step to convert from d spacing to TOF doesn't use the diffractometer constants. Following the focussing the data isn't "per detector" any more but it should still be possible to use a "per bank" DIFC\DIFA\TZERO to convert the data back to TOF
+
+There is also a more minor problem that the SaveGSS algorithm recalculates a per bank DIFC value and writes it into a header for each bank in the .gss file. This calculation also ignores the calibration output. It seems that GSAS isn't reading this header though so this problem just creates potential confusion for anyone reading the file rather than incorrect figures in GSAS.
+
+Possible solutions
+==================
+
+We could create a "backwards" mode to the AlignDetectors algorithm that performs the d spacing to TOF unit conversion including the diffractometer constants. This would then be called instead of ConvertUnits near the end of the reduction workflow.
+
+This still leaves the burden on the user\developer to remember to load the calibration data as an input parameter at two points in the workflow. Since this information has already been loaded once it seems like it should be stored in the workspace somewhere. Also the algorithm name "AlignDetectors" doesn't obviously suggest it does a unit conversion. 
+
+So on balance we prefer to put the new functionality in the ConvertUnits algorithm which is a more obvious place.
+
+Proposed Solution
+=================
+
+Modify ConvertUnits so it can incorporate diffractometer constants in its unit conversion from TOF to d spacing and vice versa.
+
+Store the "per detector" diffractometer constants in the instrument parameter map once they are applied to a workspace. A similar approach is already used to store the efixed values for indirect instruments (see ConvertUnits::getDetectorValues). This would mean that if a workspace is saved to a Nexus file, the diffractometer constants would also be saved.
+
+Create a new algorithm to load and store the diffractometer constants in the workspace called ApplyDiffCal. This would take one of three inputs:

```

Comment:
All 3 of these should, at least internally, get converted to [diffraction calibration workspace](https://docs.mantidproject.org/nightly/concepts/DiffractionCalibrationWorkspace.html).